main.py

```diff
@@ -353,10 +459,42 @@ def update_user():
     user_manipulator.update_user_data('test_user')
     return 'user updated'
 
+
 # end - actions methods #
 
 # help functions #
 
+def update_json_file(data: dict, schema_name: str, target_file_path: str, current_user_id: str, full_file: bool):
+    """
+    Try to Update json file with recived data. If full-file, re write the file. else: add or change only specific data
+    inside the json file.
+    If the data in incorrect format or not contain all needed info, return error.
+    note: when full file is false, data must contain at least one key.
+    You cant change specific inner sub key inside json file.
+    :param data: Json data received from user in dict type.
+    :param schema_name: key to untils/validators json scheme file. see json_validator.py
+    :param target_file_path: target path of the user's local json file needed update.
+    :param current_user_id:
+    :param full_file: true if full file needs to re-write.
+    :return: status.
+    """
+    try:
+        # check if the new data correct and contains all needed data
+        JsonValidator.validates(schema_name, data)
+        # add the new data to current_user json file.
+        folder_path = User.get_user_folder_path_by_id(id=current_user_id)
+        if full_file:
+            FileHandler.write_to_json(json_data=data, path=folder_path + target_file_path)
+        else:
+            FileHandler.append_to_json(data_obj_to_append=data, path=folder_path + target_file_path)
+        return jsonify({"status": 200})
+    except ValidationError as validation_error:
+        return jsonify({"Error": validation_error.message}), 400
+    except Exception as e:
+        print(e)
+        return jsonify({"Error": "Server internal error, check server"}), 400
```

Comment:
return the info inside "e" in the json you are sending to the user

Reply:
Fixed in the last version to return str(e)